smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -376,25 +909,32 @@ def process_manifest(self, manifest_path: str) -> None:
         if not self._launcher:
             raise SmartSimError(f"Unable to set launcher from {manifest_path}")
 
-        runs = [run for run in manifest.runs if run.timestamp not in self._tracked_runs]
-
+        # filter out items previously tracked and anything from separate experiments
+        runs = [
+            run
+            for run in manifest.runs
+            if run.timestamp not in self._tracked_runs and run.exp_id == self._exp_id
+        ]
         exp_dir = pathlib.Path(manifest_path).parent.parent.parent
 
         for run in runs:
             for entity in run.flatten(
-                filter_fn=lambda e: e.key not in self._tracked_jobs and e.is_managed
+                filter_fn=lambda e: e.key not in self._tracked_jobs
```

Comment:
We now need to pass the unmanaged tasks through this loop in order to instantiate all the necessary collectors.

`Line 941` still ensures we don't try to get updates for them with the JobManager instance, but then passes `False` on `line 947` because I bastardized usage of the JobManager. I can't allow my local job manager to _start_ a job, I only want it to monitor it. This tricks the JM into doing my bidding.

Reply:
Enhanced w/explanatory comment to clarify:

```py
if entity.is_managed:
      # Tell JobManager the task is unmanaged when adding so it will
      # monitor it but not try to start it
      self.job_manager.add_job(
          entity.name,
          entity.task_id,
          entity,
          False,
      )
      self._launcher.step_mapping.add(
          entity.name, entity.step_id, entity.task_id, entity.is_managed
      )
```

Reply:
> This tricks the JM into doing my bidding.

LMAO, gotcha!! That makes sense! Thanks for adding that explanatory comment! That makes much more sense on my end!